docs/data/joy/customization/themed-components/themed-components.md

```diff
@@ -94,7 +112,113 @@ extendTheme({
 });
 ```
 
-## Different styles per mode
+### Extend sizes
+
+Apply the styles when `ownerState.size` matches the new size. It is recommended to follow the existing T-shirt scale for consistency.
+
+For example, adding `xs` and `xl` sizes to the button would be:
+
+```js
+extendTheme({
+  components: {
+    JoyButton: {
+      styleOverrides: {
+        root: ({ ownerState, theme }) => ({
+          ...(ownerState.size === 'xs' && {
+            // smallest size
+            '--Icon-fontSize': '1rem',
+            '--Button-gap': '0.25rem',
+            minHeight: 'var(--Button-minHeight, 1.75rem)',
+            fontSize: theme.vars.fontSize.sm,
+            paddingBlock: '2px',
+            paddingInline: '0.5rem',
+          }),
+          ...(ownerState.size === 'xl' && {
+            // largest size
+            '--Icon-fontSize': '2rem',
+            '--Button-gap': '1rem',
+            minHeight: 'var(--Button-minHeight, 4rem)',
+            fontSize: theme.vars.fontSize.lg,
+            paddingBlock: '0.5rem',
+            paddingInline: '2rem',
+          }),
+        }),
+      },
+    },
+  },
+});
+```
+
+:::info
+**TypeScript**
+
+You need module augmentation to include the values to the `size` prop of the component.
+
+The interface is in `{ComponentName}PropsSizeOverrides` format which exists in all Joy UI components:
+
+```tsx
+// This part could be declare in your theme file
+declare module '@mui/joy/Button' {
+  interface ButtonPropsSizeOverrides {
+    xs: true;
+    xl: true;
+  }
+}
+
+// typed-safe
+<Button size="xs" />
+<Button size="xl" />
+```
+
+:::
+
+### Extend variants
+
+When you add new variant to a component, you are opting-out of the global variant feature which lets you take full control of the CSS properties related to color (`color`, `background` and `border`).
+
+```js
+extendTheme({
+  components: {
+    JoySheet: {
+      styleOverrides: {
+        root: ({ ownerState, theme }) => ({
+          ...(ownerState.variant === 'glass' && {
+            color: theme.vars.palette.text.primary,
+            background: 'rgba(255, 255, 255, 0.14)',
+            backdropFilter: 'blur(5px)',
+            border: '1px solid rgba(255, 255, 255, 0.3)',
+            boxShadow: '0 4px 30px rgba(0, 0, 0, 0.1)',
+          }),
+        }),
+      },
+    },
+  },
+});
+```
+
+:::info
+
+**TypeScript**
+
+You need module augmentation to include the values to the `variant` prop of the component.
```

Comment:
```suggestion
ℹ️ **TypeScript tip**:  You need module augmentation to include the values to the `size` prop of the component.
```

Reply:
Actually, this is needed rather than a tip. Should I move it out of the `:::info`?

Reply:
Ah, that makes sense. Yeah, I think so!